integration/agent/internal/src/main/java/com/splunk/rum/integration/agent/internal/processor/SplunkInternalGlobalAttributeSpanProcessor.kt

```diff
@@ -54,5 +54,12 @@ class SplunkInternalGlobalAttributeSpanProcessor : SpanProcessor {
 
     companion object {
         val attributes: MutableList<Attribute> = ArrayList()
+
+        fun updateStringAttribute(key: String, value: String?) {
```

Comment:
Nit: This might be useful in multiple places and use cases. Consider extracting it into a utility method.

Idea:
```koltin
fun MutableList<Attribute>.updateStringAttribute(key: String, value: String) {
    this.removeIf { it.name == key }
    this += Attribute.String(key, value)
}
```

Reply:
Nit: Also, so you don't have to always "convert" between `key: String` and `key: AttributeKey<String>`, it can be beneficial to have this conversion resolved by defining two functions. For example:

```
fun MutableList<Attribute>.updateStringAttribute(key: AttributeKey<String>, value: String) = 
    updateStringAttribute(key.key, value)

fun MutableList<Attribute>.updateStringAttribute(key: String, value: String) {
    this.removeIf { it.name == key }
    this += Attribute.String(key, value)
}
```